sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -124,11 +153,24 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
     val hiveVersion = externalCatalog.unwrapped.asInstanceOf[HiveExternalCatalog].client.version
     val stagingDir = hadoopConf.get("hive.exec.stagingdir", ".hive-staging")
     val scratchDir = hadoopConf.get("hive.exec.scratchdir", "/tmp/hive")
+    logDebug(s"path '${path.toString}', staging dir '$stagingDir', " +
+      s"scratch dir '$scratchDir' are used")
 
     if (hiveVersionsUsingOldExternalTempPath.contains(hiveVersion)) {
       oldVersionExternalTempPath(path, hadoopConf, scratchDir)
     } else if (hiveVersionsUsingNewExternalTempPath.contains(hiveVersion)) {
-      newVersionExternalTempPath(path, hadoopConf, stagingDir)
+      // HIVE-14270: Write temporary data to HDFS when doing inserts on tables located on S3
```

Comment:
From the config doc, this feature is only supported in Hive 2.0+. But `hiveVersionsUsingNewExternalTempPath` also covers Hive 1.1, 1.2.

Reply:
Added version check using `hiveVersionsSupportingNonBlobstoreTempPath`.

```
    val hiveVersionsSupportingNonBlobstoreTempPath: Set[HiveVersion] =
      Set(v2_0, v2_1, v2_2, v2_3, v3_0, v3_1)
```